python/ml4ir/base/model/architectures/dnn.py

```diff
@@ -60,13 +67,21 @@ def get_op(layer_type, layer_args):
         ]
 
     def get_architecture_op(self):
-        def _architecture_op(ranking_features):
+        def _architecture_op(ranking_features, metadata_features):
             layer_input = ranking_features
 
             # Pass ranking features through all the layers of the DNN
             for layer_op in self.layer_ops:
                 layer_input = layer_op(layer_input)
 
+            if 'positional_bias_handler' in self.model_config and \
+                    self.model_config['positional_bias_handler']['key'] == \
+                    PositionalBiasHandler.FIXED_ADDITIVE_POSITIONAL_BIAS:
+
+                positional_bias = self.positional_bias_layer(metadata_features[self.feature_config.get_rank()['name']],
+                                                             max_ranks=self.model_config['positional_bias_handler']['max_ranks_count'], training=True)
```

Comment:
Ooooh found a bug after breaking my head over this for a long time. You are hardcoding `training=True` here. You shouldn't. Tensorflow/keras will automatically send the right value when you call `fit()` vs `predict()`. What this does is basically hard set it to training=True, regardless of training vs inference - dangerous.